### PR TITLE
[HINTS] Add implementation for hint on finalize_blake2s (cario_blake2s)

### PR DESCRIPTION
**Pull request by fmoletta**
Add implementation for hint on finalize_blake2s (cairo_blake2s.cairo)
Add unit tests
Add integration test

### CODE REVIEW

**Comment by Oppen** [comment]
It'll make the review much easier if you rebase against main. Otherwise I'll be lost in duplicate changes from the older branch D:

File: src/vm/hints/blake2s_utils.rs [hunk 1]
```diff
@@ -90,6 +91,58 @@ pub fn compute_blake2s(
...
+        t1=0,
+        f0=0xffffffff,
+        f1=0,
+    )
+    padding = (modified_iv + message + [0, 0xffffffff] + output) * (_n_packed_instances - 1)
+    segments.write_arg(ids.blake2s_ptr_end, padding)
+*/
+pub fn finalize_blake2s(
+    vm: &mut VirtualMachine,
+    ids: HashMap<String, BigInt>,
```
Comment (Oppen) [inline src/vm/hints/blake2s_utils.rs:118]:
```suggestion
    ids: &HashMap<String, BigInt>,
```

File: src/vm/hints/blake2s_utils.rs [hunk 2]
```diff
@@ -90,6 +91,58 @@ pub fn compute_blake2s(
...
+    modified_iv[0] = IV[0] ^ 0x01010020;
+    let output = blake2s_compress(&modified_iv, &message, 0, 0, 0xffffffff, 0);
+    let mut padding = modified_iv.to_vec();
+    padding.extend(message);
+    padding.extend([0, 0xffffffff]);
+    padding.extend(output);
+    let mut full_padding = Vec::<u32>::new();
+    for _ in 0..N_PACKED_INSTANCES - 1 {
+        full_padding.extend(padding.clone());
+    }
```
Comment (Oppen) [inline src/vm/hints/blake2s_utils.rs:134]: You can probably reduce cloning and reallocations in this loop:
```suggestion
    let padding = padding.as_slice();
    let mut full_padding = Vec::<u32>::with_capacity(padding.len() * N_PACKED_INSTANCES);
    for _ in 0..N_PACKED_INSTANCES - 1 {
        full_padding.extend_from_slice(padding);
    }
```

**Review by Oppen — CHANGES_REQUESTED** [review]

File: src/vm/hints/blake2s_utils.rs (hunk 2 above)
Comment (fmoletta) [reply src/vm/hints/blake2s_utils.rs]: Fixed!

**Review by Oppen — APPROVED** [review]

**Pull request merged**